commons/client/ServiceInstance.py

```diff
@@ -0,0 +1,90 @@
+# standard library
+from abc import ABC, abstractmethod
+from urllib.parse import urlparse
+
+__author__ = "Waterball (johnny850807@gmail.com)"
+__license__ = "Apache 2.0"
+
+
+class ServiceInstance(ABC):
+    @property
+    @abstractmethod
+    def instance_id(self):
+        pass
```

Comment:
In your class diagram, instance_id is a private attribute, but you make it an abstract getter in your implementation. Why don't you just make instance_id a private attribute in ServiceInstance and get rid of the abstract getter? What is your intent to make an abstract getter without declaring an attribute first?

Reply:
Since it's an **attribute** of an **interface** as the class diagram says.

Reply:
If you have no more questions, you must approve the PR.